src/clr-angular/utils/focus-trap/focus-trap.directive.ts

```diff
@@ -67,10 +103,18 @@ export class FocusTrapDirective implements AfterViewInit, OnDestroy {
     if (isPlatformBrowser(this.platformId) && this.focusTrapsTracker.nbFocusTrappers === 1) {
       this.topReboundEl = this.createFocusableOffScreenEl();
       this.bottomReboundEl = this.createFocusableOffScreenEl();
-      // Add reboundBeforeTrapEl to the document body as the first child
-      this.renderer.insertBefore(this.document.body, this.topReboundEl, this.document.body.firstChild);
-      // Add reboundAfterTrapEl to the document body as the last child
-      this.renderer.appendChild(this.document.body, this.bottomReboundEl);
+      if (this.config.local) {
+        const hostElement = this.el.nativeElement;
+        // Add reboundBeforeTrapEl to document right before the host element
+        this.renderer.insertBefore(hostElement.parentElement, this.topReboundEl, hostElement);
+        // Add reboundAfterTrapEl to the document right after host element
+        this.renderer.appendChild(hostElement, this.bottomReboundEl);
+      } else {
+        // Add reboundBeforeTrapEl to the document body as the first child
+        this.renderer.insertBefore(this.document.body, this.topReboundEl, this.document.body.firstChild);
+        // Add reboundAfterTrapEl to the document body as the last child
+        this.renderer.appendChild(this.document.body, this.bottomReboundEl);
+      }
```

Comment:
If we actually put the rebounder elements inside of the host element, they get cleaned up automatically for us. Otherwise, I'm having to do a lot of work to seek out the elements because by onDestroy the template is disconnected from the DOM and I can't access the parent to remove the child nodes. I think this actually works better, contains the manipulation to the local area better, and cleans up nicely.

Reply:
Actually without support for shift+tab, we don't need a top rebounder here at all because there is no case where the top rebounder will do anything.